### ISSUE TITLE
Search in Thaumonomicon doesn't work.

### ISSUE THREAD

**Issue by OneEyeMaker** [BluSunrize/WitchingGadgets#87]
Hello.
I recently began to learn this mod and maybe I don't know something... But the main feature, that players notice imediately - Thaumonomicon search - doesn't work for me. I see search bar in book, but it isn't clickable at all.
Is this feature disabled for now?

**Comment by BluSunrize** [comment]
Nope, it's not disabled, it should work :/

**Comment by OneEyeMaker** [comment]
Strange... I can't click in text box!

**Comment by BluSunrize** [comment]
Please be so kind as to provide a screenshot. I also can't reproduce this, so I have to assume that it's another mod messing with it
I'd also like to know what GUI scale you use

**Comment by OneEyeMaker** [comment]
Tested just now only with: 
Forge 1448, NEI (+core+addons), Thaumcraft (+Baubles) and Witching Gadgets (+Traveller's Gear) and FastCraft (without this mod I have 10 fps and less).
I checked this problem in all GUI scales.
Result: I can't click in text box at all in any GUI scale.
http://imgur.com/a/6ls6D

**Comment by BluSunrize** [comment]
Can't reproduce this, but I noticed a slight delay. try holding the button or double-clicking

**Comment by Alaberti** [comment]
I can confirm I had this issue in 1.1.9 (have not tested the latest) and it is intermittent. Often I am unable to click the text box, but occasionally I can. This is in a modded environment. I will run some tests, see if it happens if only certain mods are in or something weird like that.

**Comment by OneEyeMaker** [comment]
Holding mouse button and double-clicking don't work for me too.

**Comment by ghost** [comment]
@BluSunrize  I believe it may be an optifine issue

@OneEyeMaker  do you use Optifine? If you do try playing around with Fullscreen YES NO / Resolution . Seems to work for me if I fiddle with that setting and restart a few times.

**Comment by OneEyeMaker** [comment]
I never use Optifine. With my _integrated_ graphics card Optifine doesn't provide tangible improvements; it only increases the load.
I'm only using FastCraft for improving FPS (on client side) and TPS (on server side). 

But I tested this bug **with and without** FastCraft in all GUI scales in both fullscreen and window mode. Result: I can't activate search bar (I can't type there).

P. S.
@BluSunrize 
In ImmersiveEngineering 0.6+ you also added search bar in book. As I understand, It should activate by click and after changing page. In first case (by click) I can't activate it. But it normally works, when I go to contents of manual or contents of any chapter.
In that situation, maybe you should add another way to activate search bar in Thaumonomicon (keybinding?).

**Comment by BluSunrize** [comment]
I was already considering making it active by default since no typing is required in the thaumonomicon anyway, so there's nothing wrong with having it perma-active. But once again: I'm /not working/ on WG atm.

**Comment by ghost** [comment]
@BluSunrize  making it perma active without having to click will fix all bugs and issues instantly

**Comment by ghost** [comment]
@BluSunrize  i agree about optifine. it also conflicts with a bunch of other mods for example bibliocraft. i only use it cause my computer is pretty slow and it does add about 20 or 30 fps no kidding. that plus fastcraft almost triples my fps and even then i lag occasionally. if i could i'd use shaders and fancy graphics but for now i'll stick with optifine. I understand and I agree. The above suggestion permanently enabled Search so any time you open Thaumonomicon you can start typing should work fine except pressing E (or Minecraft Use) might close Thaumonomicon so you need to disable that and use escape instead . I dunno. 

edit: I am pretty sure Automagy does this for their requision tome where you tell golems to bring you things. The moment you open the tome you can start typing and pressing ESC exists the gui.

**Comment by Teteros** [comment]
@BluSunrize Same issue here.
May I suggest enabling the search feature when the focus ("F") key is pressed, just like NEI?
This may be the most intuitive to the user, as many of us use it, or an alternative CTRL+ Enter (project red) or a custom key in options.
Running optifine for reference, yes I know this may be the cause, but the hotkey is a workaround anyone can work with.
